File: scripts/get_latest_changes.py
# Copyright 2021-2024 Aidan Sun and the WhaleConnect contributors
# SPDX-License-Identifier: GPL-3.0-or-later

# This script is used in the CI to publish releases

from pathlib import Path
import re
import sys

changelog_path = project_dir = Path(__file__).parent.parent / "docs" / "changelog.md"

with open(changelog_path, "r") as f:
    changelog = f.read()

base_text = """These notes are also available on the [changelog](https://github.com/WhaleConnect/whaleconnect/blob/main/docs/changelog.md).

All executables are unsigned, so you may see warnings from your operating system if you run them.

"""

if latest_changes := re.search(r"\n## .+?\n\n(.+?)\n\n## ", changelog, re.S):
    text = base_text + latest_changes.group(1)
    with open("changelog-latest.txt", "w") as f:
        f.write(text)
else:
    sys.exit(1)

File: xmake/scripts/get_latest_changes.lua
-- Copyright 2021-2024 Aidan Sun and the WhaleConnect contributors
-- SPDX-License-Identifier: GPL-3.0-or-later

-- This script is used in the CI to publish releases

local baseText = [[These notes are also available on the [changelog](https://github.com/WhaleConnect/whaleconnect/blob/main/docs/changelog.md).

All executables are unsigned, so you may see warnings from your operating system if you run them.

]]

function main(...)
    local changelogPath = path.join(os.projectdir(), "docs", "changelog.md")
    local changelog = io.readfile(changelogPath)

    local latestChanges = changelog:match("\n## .-\n\n(.-)\n\n## ")
    if latestChanges then
        io.writefile("changelog-latest.txt", baseText .. latestChanges)
    else
        os.exit(1)
    end
end
